tests/test_barostat.py

```diff
@@ -545,3 +545,89 @@ def test_get_group_indices():
     with pytest.raises(AssertionError):
         # num_atoms <  an atom's index in bond_idxs
         get_group_indices([[0, 3]], num_atoms=3)
+
+
+@pytest.mark.memcheck
+def test_barostat_scaling_behavior():
+    """Verify that it is possible to retrieve and set the volume scaling factor. Also check that the adaptive behavior of the scaling can be disabled"""
+    lam = 1.0
+    temperature = DEFAULT_TEMP
+    timestep = 1.5e-3
+    barostat_interval = 3
+    collision_rate = 1.0
+    seed = 2021
+    np.random.seed(seed)
+
+    pressure = DEFAULT_PRESSURE
+
+    mol_a, _, _ = get_hif2a_ligand_pair_single_topology()
+    ff = Forcefield.load_from_file("smirnoff_1_1_0_sc.py")
+
+    host_system, host_coords, host_box, host_top = build_water_system(3.0, ff.water_ff)
+    bt = BaseTopology(mol_a, ff)
+    afe = AbsoluteFreeEnergy(mol_a, bt)
+    host_config = HostConfig(host_system, host_coords, host_box, host_coords.shape[0])
+    unbound_potentials, sys_params, masses = afe.prepare_host_edge(ff.get_params(), host_config, lam)
+    coords = afe.prepare_combined_coords(host_coords=host_coords)
+
+    # get list of molecules for barostat by looking at bond table
+    harmonic_bond_potential = unbound_potentials[0]
+    bond_list = get_bond_list(harmonic_bond_potential)
+    group_indices = get_group_indices(bond_list, len(masses))
+
+    u_impls = []
+    for params, unbound_pot in zip(sys_params, unbound_potentials):
+        bp = unbound_pot.bind(params)
+        bp_impl = bp.to_gpu(precision=np.float32).bound_impl
+        u_impls.append(bp_impl)
+
+    integrator = LangevinIntegrator(
+        temperature,
+        timestep,
+        collision_rate,
+        masses,
+        seed,
+    )
+
+    v_0 = sample_velocities(masses, temperature)
+
+    baro = custom_ops.MonteCarloBarostat(
+        coords.shape[0],
+        pressure,
+        temperature,
+        group_indices,
+        barostat_interval,
+        u_impls,
+        seed,
+    )
+    # Initial volume scaling is 0
+    assert baro.get_volume_scaling() == 0.0
```

Comment:
nit[non-corner case as default]: preference for this to be a non-zero value.

(I believe OpenMM's MonteCarloBarostat initializes with something like `proposal_scale = 0.01 * volume(initial_box)` or similar.)

Reply:
It does get initialized to that value, but when you construct the barostat it doesn't know what the volume is. 

https://github.com/proteneer/timemachine/blob/master/timemachine/cpp/src/kernels/k_barostat.cuh#L87-L89

Reply:
Ahh, thanks for clarifying!

(Less confident: but may be useful for this to always be positive, even at initialization (maybe 1% of `volume(4A x 4A 4A solvent box)`?))

Reply:
What sort of case are you imagining this to apply? If it was some default value, it could be significantly off and would probably need to be re-initialized and you would want to reinitialize. For example, if it was set for a 4A box, and you gave it a 10A box it seems like it could take significantly longer to reach equilibrium.

Reply:
Sorry, I was just confused.

(Wanted to avoid the case where a proposal scale of 0.0 was ever used. Reading more carefully, although the proposal scale can _nominally_ be 0.0, it will never _effectively_ be 0.0. Enforcing that this value is nominally always positive -- while retaining the logic "the first time you see an input configuration, reset proposal scale to 1% of its volume" -- wouldn't actually do anything different than the current implementation.)